### PR TITLE
Moved debug error to variable position lookup

### PR DESCRIPTION
**Pull request by msandre**
FastGetSolutionStepValue was not throwing an error in debug when the variable was missing on the node. We decided to move the error check to the Position function to ensure all getter methods throw a consistent error.

### CODE REVIEW

File: kratos/containers/variables_list_data_value_container.h
Comment (jcotela) [inline kratos/containers/variables_list_data_value_container.h:336]: Here we woud check for the variable name twice (once in GetValue, once in Position)

File: kratos/containers/variables_list_data_value_container.h
Comment (jcotela) [inline kratos/containers/variables_list_data_value_container.h:350]: Here we woud check for the variable name twice (once in GetValue, once in Position)

File: kratos/containers/variables_list_data_value_container.h
Comment (jcotela) [inline kratos/containers/variables_list_data_value_container.h]: Here Position will acces the database by index, so no checks are done. I'd leave the error macros in this case.

File: kratos/containers/variables_list_data_value_container.h
Comment (msandre) [reply kratos/containers/variables_list_data_value_container.h:336]: Yes, in debug the GetSolutionStepValue may be slower. If we want to avoid this, we need to go back to copying the error checks to each function like before. I can change it if you prefer that way.

Comment (jcotela) [reply kratos/containers/variables_list_data_value_container.h:336]: I don't think the method being slower in debug is really a problem, I was just commenting because I believe that the check in line 335 is the same as that in 971, which would be called from here.

Comment (msandre) [reply kratos/containers/variables_list_data_value_container.h:336]: There is a difference since like 335 is checking with or without debug. This seems to be the only difference between GetValue and FastGetValue.

Comment (jcotela) [reply kratos/containers/variables_list_data_value_container.h:336]: Oh you are right... I did not realize this macro is not debug-only. Disregard the comment then.

**Review by RiccardoRossi — COMMENTED** [review]
i think this can be approved. @jcotela i leave this to you since you looked into it in greater depth

**Review by jcotela — APPROVED** [review]

**Review by pooyan-dadvand — COMMENTED** [review]
Do we need to repeat the checks in `GetValue` and  `Position` methods? I think we only need to do it in `Position`.

To be honest, Now I would drop the GetValue behaviour and only keep the debug errors reducing number of redundant methods. @RiccardoRossi ?

**Comment by msandre** [comment]
@pooyan-dadvand we need to repeat the checks to have the proper behavior in debug and release mode. I will leave the choice of dropping GetValue to you, but would suggest we do it in a different pr. I will merge this in the next day if there are no more comments. Thx

**Pull request merged**
